site2/docs/functions-package.md

```diff
@@ -0,0 +1,524 @@
+    ```text
+    ...
+    07:55:03.724 [main] INFO  org.apache.pulsar.functions.runtime.ProcessRuntime - Started process successfully
+    ...
+    ```
+
+    > **Tip**
+    >
+    >  - For the description about the parameters (for example, `--classname`, `--py`, `--inputs`, and so on), run the command `./bin/pulsar-admin functions` to get more information or see [here](http://pulsar.apache.org/docs/en/pulsar-admin/#functions).
+    > 
+    > - If you want to start a function in cluster mode, replace `localrun` with `create` in the command above. The following log indicates that the Python function starts up successfully.
+    >
+    >   ```text
+    >   "Created successfully"
+    >   ```
+
+### ZIP file
+
+This example demonstrates how to package a function by **ZIP file** in Python.
+
+> **Note**
+>
+> This example assumes that you have [run a standalone Pulsar in Docker](#Run a standalone Pulsar in Docker) successfully.
+
+1. Prepare the ZIP file
+
+When packaging the ZIP file of the Python Function, the following requirements need to be met:
+
+```text
+Assuming zip file with format `func.zip`, extract to folder function and internal dir format:
+    "func/src"
+    "func/requirements.txt"
+    "func/deps"
+```
+Now we take [exclamation.zip](https://github.com/apache/pulsar/tree/master/tests/docker-images/latest-version-image/python-examples) as an example, its internal structure is as follows:
+
+```text
+.
+├── deps
+│   └── sh-1.12.14-py2.py3-none-any.whl
+└── src
+    └── exclamation.py
+```
+
+2. Run the Python Function
+
+    (1) Copy the ZIP file to the Pulsar image.
+
+    ```bash
+    docker exec -it [CONTAINER ID] /bin/bash
+    docker cp <path of ZIP file>  CONTAINER ID:/pulsar
+    ```
+
+    (2) Run the Python function using the following command.
+
+    ```bash
+    ./bin/pulsar-admin functions localrun \
+    --classname exclamation \
+    --py <path of ZIP file> \
+    --inputs persistent://public/default/in-topic \
+    --output persistent://public/default/out-topic \
+    --tenant public \
+    --namespace default \
+    --name PythonFunction
+    ```
+
+    The following log indicates that the Python function starts up successfully.
+
+    ```text
+    ...
+    07:55:03.724 [main] INFO  org.apache.pulsar.functions.runtime.ProcessRuntime - Started process successfully
+    ...
+    ```
+
+    > **Tip**
+    >
+    >  - For the description about the parameters (for example, `--classname`, `--py`, `--inputs`, and so on), run the command `./bin/pulsar-admin functions` to get more information or see [here](http://pulsar.apache.org/docs/en/pulsar-admin/#functions).
+    > 
+    > - If you want to start a function in cluster mode, replace `localrun` with `create` in the command above. The following log indicates that the Python function starts up successfully.
+    >
+    >   ```text
+    >   "Created successfully"
+    >   ```
+
+### PIP
+
+This example demonstrates how to package a function by **PIP** in Python.
+
+> **Note**
```

Comment:
If you have more than one information under a note, please use unordered list. If the unordered list is not used, please make sure there are two spaces at the end of a line, so the next line is a new line.